new-log-viewer/src/services/LogFileManager/index.ts

```diff
@@ -223,42 +203,56 @@ class LogFileManager {
             currentLine += msg.split("\n").length - 1;
         });
 
+        const newPageNum: number = getChunkNum(beginLogEventNum, this.#pageSize);
+
         return {
-            logs: messages.join(""),
             beginLineNumToLogEventNum: beginLineNumToLogEventNum,
             cursorLineNum: 1,
+            logEventNum: newLogEventNum,
+            logs: messages.join(""),
+            pageNum: newPageNum,
         };
     }
 
     /**
-     * Gets the range of log event numbers for the page containing the given cursor.
+     * Gets the range of log event numbers for the page and the new log event number.
      *
      * @param cursor The cursor object containing the code and arguments.
-     * @return The range.
+     * @return The range and new log event number.
      * @throws {Error} if the type of cursor is not supported.
      */
-    #getCursorRange (cursor: CursorType): { beginLogEventNum: number, endLogEventNum: number } {
-        if (0 === this.#numEvents) {
-            return {
-                beginLogEventNum: 1,
-                endLogEventNum: 0,
-            };
-        }
-
-        let beginLogEventIdx: number = 0;
+    #getCursorData (cursor: CursorType): {
+        beginLogEventNum: number,
+        endLogEventNum: number,
+        newLogEventNum: number
```

Comment:
Can we rename these to:

* pageBeginLogEventNum
* pageEndLogEventNum
* matchingLogEventNum

Since the return value is no longer *just* a range, I think we should be more explicit with the names.